Branch 1.7.x build failed due to missing camel-bom transient dependency

## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description
Branch 1.7.x build is failing due to the following error:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Integration 1.7-SNAPSHOT ........................... SUCCESS [  6.101 s]
[INFO] Integration :: Support :: Camel Acme ............... SUCCESS [ 15.613 s]
[INFO] Integration :: API ................................. SUCCESS [ 11.791 s]
[INFO] Integration :: Component Proxy ..................... SUCCESS [ 16.599 s]
[INFO] Integration :: Runtime ............................. SUCCESS [ 23.971 s]
[INFO] Integration :: Bill of Materials (BOM) ............. SUCCESS [  0.149 s]
[INFO] Integration :: Project Generator ................... SUCCESS [ 16.465 s]
[INFO] Integration :: Runtime :: Camel-k .................. FAILURE [ 11.290 s]
[INFO] Integration :: Camel K :: Bill of Materials (BOM) .. SKIPPED
[INFO] Integration :: Runtime :: SpringBoot 1.7-SNAPSHOT .. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:43 min
[INFO] Finished at: 2019-07-05T15:56:10-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin org.apache.camel.k:camel-k-maven-plugin:0.3.4.fuse-740007 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.camel.k:camel-k-maven-plugin:jar:0.3.4.fuse-740007: Could not find artifact org.apache.camel:camel-bom:pom:2.21.0.fuse-740037 in redhat-ga-repository (https://maven.repository.redhat.com/ga) -> [Help 1]
```

In the app/pom.xml, `camel-k-runtime.version` property is set to `0.3.4.fuse-740007`:
https://github.com/syndesisio/syndesis/blob/2a5af41ba0cd6639e6456f8c5cdd71e6f2d260c8/app/pom.xml#L81

`camel-k-runtime-parent` version `0.3.4.fuse-740007` depends on Camel version `2.21.0.fuse-740037`:
https://repository.jboss.org/nexus/content/groups/ea/org/apache/camel/k/camel-k-runtime-parent/0.3.4.fuse-740007/camel-k-runtime-parent-0.3.4.fuse-740007.pom

Camel version `2.21.0.fuse-740037` has been purged in the Maven repository:
https://repository.jboss.org/nexus/content/groups/ea/org/apache/camel/camel-bom/2.21.0.fuse-740037/

The only valid version of `camel-k-runtime-parent` seems to be `0.3.4.fuse-740001`, which references Camel version `2.21.0.fuse-740028`.

--- a/app/extension/bom/pom.xml
+++ b/app/extension/bom/pom.xml
@@ -28,7 +28,7 @@
 
   <properties>
     <spring-boot.version>1.5.13.RELEASE</spring-boot.version>
-    <camel.version>2.21.0.fuse-740028</camel.version>
+    <camel.version>2.21.0.fuse-740038</camel.version>
   </properties>
 
     <!-- Metadata need to publish to central -->

--- a/app/integration/bom-camel-k/pom.xml
+++ b/app/integration/bom-camel-k/pom.xml
@@ -27,7 +27,7 @@
   <packaging>pom</packaging>
 
   <properties>
-    <camel.version>2.21.0.fuse-740028</camel.version>
+    <camel.version>2.21.0.fuse-740038</camel.version>
     <atlasmap.version>1.39.5</atlasmap.version>
     <jackson.version>2.8.11</jackson.version>
     <jackson.databind.version>2.8.11.3</jackson.databind.version>

--- a/app/integration/bom/pom.xml
+++ b/app/integration/bom/pom.xml
@@ -28,7 +28,7 @@
 
   <properties>
     <spring-boot.version>1.5.13.RELEASE</spring-boot.version>
-    <camel.version>2.21.0.fuse-740028</camel.version>
+    <camel.version>2.21.0.fuse-740038</camel.version>
     <atlasmap.version>1.40.2</atlasmap.version>
     <jackson.databind.version>2.8.11.3</jackson.databind.version>
   </properties>

--- a/app/pom.xml
+++ b/app/pom.xml
@@ -77,8 +77,8 @@
     <commons.compress.version>1.18</commons.compress.version>
 
     <!-- Global camel version used everywhere -->
-    <camel.version>2.21.0.fuse-740028</camel.version>
-    <camel-k-runtime.version>0.3.4.fuse-740007</camel-k-runtime.version>
+    <camel.version>2.21.0.fuse-740038</camel.version>
+    <camel-k-runtime.version>0.3.4.fuse-740008</camel-k-runtime.version>
 
     <!-- Don't fork based on cores, doesn't work nicely in the cloud -->
     <basepom.test.fork-count>1</basepom.test.fork-count>

